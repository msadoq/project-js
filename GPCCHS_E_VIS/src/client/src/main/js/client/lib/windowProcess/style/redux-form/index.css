.redux-form-dirty {
  box-shadow: 0px 0px 1px #B55, 0px 0px 2px #B55, 0px 0px 3px #B55;
}
.redux-form-padded {
  padding: 4px;
}
