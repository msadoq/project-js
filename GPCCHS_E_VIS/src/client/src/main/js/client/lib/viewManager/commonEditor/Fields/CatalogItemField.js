import React, { PureComponent, PropTypes } from 'react';
import { Field } from 'redux-form';
import ReactSelectField from 'windowProcess/commonReduxForm/ReactSelectField';
import { catalogItemType } from 'viewManager/common/Components/types';
import { computeOptions } from 'viewManager/commonEditor/Fields/common';

const { string, arrayOf, oneOfType, func, number } = PropTypes;

export default class CatalogItemField extends PureComponent {
  static propTypes = {
    catalogItems: oneOfType([
      string,
      arrayOf(catalogItemType),
    ]),
    askCatalogItems: func.isRequired,
    sessionId: number,
    domainId: number,
    timelineId: string,
    catalogName: string,
  };

  static defaultProps = {
    catalogItems: null,
    sessionId: null,
    domainId: null,
    timelineId: null,
    catalogName: null,
  };

  componentWillReceiveProps(nextProps) {
    const {
      domainId,
      timelineId,
      sessionId,
      catalogItems,
      askCatalogItems,
      catalogName,
    } = nextProps;

    if (!!(domainId && timelineId && catalogName) && catalogItems === null) {
      askCatalogItems(domainId, sessionId, catalogName);
    }
  }

  render() {
    const { catalogItems, domainId, timelineId, catalogName } = this.props;
    const disabled = (!domainId || !timelineId || !catalogName || catalogItems === null);
    return (
      <Field
        format={null}
        name="catalogItem"
        component={ReactSelectField}
        clearable
        disabled={disabled}
        options={computeOptions(catalogItems)}
      />
    );
  }
}
