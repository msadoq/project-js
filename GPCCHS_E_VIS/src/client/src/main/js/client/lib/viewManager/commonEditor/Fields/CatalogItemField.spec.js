import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { reduxForm } from 'redux-form';
import CatalogItemField from 'viewManager/commonEditor/Fields/CatalogItemField';
import { shallow } from 'enzyme';

const propsStub = {
  catalogItems: [{
    name: 'CatalogItemFieldName',
    items: [],
  }],
  askCatalogItems: () => null,
  timelineId: 'timelineId',
  sessionId: 1,
  domainId: 1,
  catalogName: 'catalogName',
};
describe('CatalogItemField :: render', () => {
  test('CatalogItemField :: render', () => {
    const Decorated = reduxForm({ form: 'testForm' })(CatalogItemField);
    const store = createStore(state => state, {});
    const tree = shallow(
      <Provider store={store}>
        <Decorated
          {...propsStub}
        />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });
  test('CatalogItemField :: render requesting', () => {
    const propsStub2 = { ...propsStub, catalogs: 'requesting' };
    const wrapper = shallow(
      <CatalogItemField
        {...propsStub2}
      />
    );
    wrapper.setProps({
      domainId: 1,
      timelineId: 'timelineId',
      catalogName: 'catalogName',
      itemName: 'itemName',
      CatalogItemFields: null,
    });
    expect(wrapper).toMatchSnapshot();
  });
});
