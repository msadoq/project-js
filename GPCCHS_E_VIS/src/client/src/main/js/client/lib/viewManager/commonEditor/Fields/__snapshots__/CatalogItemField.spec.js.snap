// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CatalogItemField :: render CatalogItemField :: render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(Symbol.observable): [Function],
          }
    }
>
    <ReduxForm
        askCatalogItems={[Function]}
        catalogItems={
            Array [
                Object {
                  "items": Array [],
                  "name": "CatalogItemFieldName",
                },
              ]
        }
        catalogName="catalogName"
        domainId={1}
        sessionId={1}
        timelineId="timelineId"
    />
</Provider>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "askCatalogItems": [Function],
      "catalogItems": Array [
        Object {
          "items": Array [],
          "name": "CatalogItemFieldName",
        },
      ],
      "catalogName": "catalogName",
      "domainId": 1,
      "sessionId": 1,
      "timelineId": "timelineId",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "askCatalogItems": [Function],
        "catalogItems": Array [
          Object {
            "items": Array [],
            "name": "CatalogItemFieldName",
          },
        ],
        "catalogName": "catalogName",
        "domainId": 1,
        "sessionId": 1,
        "timelineId": "timelineId",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`CatalogItemField :: render CatalogItemField :: render requesting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CatalogItemField
    CatalogItemFields={null}
    askCatalogItems={[Function]}
    catalogItems={
        Array [
            Object {
              "items": Array [],
              "name": "CatalogItemFieldName",
            },
          ]
    }
    catalogName="catalogName"
    catalogs="requesting"
    domainId={1}
    itemName="itemName"
    sessionId={1}
    timelineId="timelineId"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clearable": true,
      "component": [Function],
      "disabled": false,
      "format": null,
      "name": "catalogItem",
      "options": Array [
        Object {
          "label": "CatalogItemFieldName",
          "value": "CatalogItemFieldName",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clearable": true,
        "component": [Function],
        "disabled": false,
        "format": null,
        "name": "catalogItem",
        "options": Array [
          Object {
            "label": "CatalogItemFieldName",
            "value": "CatalogItemFieldName",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
